sl_init_mod.f90

```diff
@@ -89,15 +89,17 @@ end subroutine check
 
 
    ! -------------------------------------------------------------------------
-   subroutine read_sl(data_slm, filename, filepath, suffix, fext)
+   subroutine read_sl(data_slm, filename, filepath, nglv, suffix, fext)
```

Comment:
Seeing these actual changes, I think it would have been cleaner and less overall work to not modify the calling signature of these i/o routines and instead just make the arrays allocatable.  But given you've already implemented this and it works, and the time crunch, I'm fine to keep it like this.

Reply:
Thanks for pointing this out. I misjudged before that I needed to pass down the `nglv` value as an argument to have everything work correctly, but it wasn't necessary at all. I've changed it back to the original calling signature of the i/o routines.